docs/usage/configuration/config_documentation.md

```diff
@@ -3612,6 +3612,11 @@ Sub-options include:
 * `jaeger_config`: Jaeger can be configured to sample traces at different rates.
    All configuration options provided by Jaeger can be set here. Jaeger's configuration is
    mostly related to trace sampling which is documented [here](https://www.jaegertracing.io/docs/latest/sampling/).
+* `request_headers_to_tag`: A list of headers to extract from the request and
+  add to to the top-level servlet tracing span as tags. Useful when you're using
+  a reverse proxy service like Cloudflare to protect your Synapse instance in
+  order to correlate and match up requests that timed out at the Cloudflare
+  layer to the Synapse traces.
```

Comment:
Not to throw a wrench in these plans, but if other reverse proxies do something similar; I wonder if it would make more sense to have a header which could be the source for `SynapseRequest.request_id` (instead of generating a new one):

https://github.com/matrix-org/synapse/blob/9772e362aaef0740039662e3dc93a0de1411eb06/synapse/http/site.py#L174-L175

Would that be clearer instead of having two separate IDs?

Reply:
👍 Seems reasonable to me and a little cleaner. I'll create a new PR with this solution ⏩

Reply:
I don't know if this would make it harder to track things in the logs or anything, but seems like it would be a lot cleaner.

Reply:
Tried this out in https://github.com/matrix-org/synapse/pull/13801

Losing the simplicity of the incrementing integer to reference in logs as you scroll kinda sucks though. I'm leaning back to this option I think 🤔

Reply:
> Losing the simplicity of the incrementing integer to reference in logs as you scroll kinda sucks though. I'm leaning back to this option I think 🤔

Is that really used though? I usually just copy and paste and then grep. Once the number gets beyond a few digits it is useless anyway IMO.

Reply:
I'm not the one having to log dive and debug big Synapse instances so I don't know. In tests, we will still have the same sequential request ID's since this config won't be defined. We're not changing any default either so it's basically opt-in.

If anyone has a strong desire for the old way, we can always revert as well. If you're happy with https://github.com/matrix-org/synapse/pull/13801, we can move forward with that ⏩

Reply:
Sounds like we should ask folks real quick then!

Reply:
Thanks for asking in #synapse-dev, https://matrix.to/#/!vcyiEtMVHIhWXcJAfl:sw1v.org/$kMyUMCOU4fZpH6AOUPOIqGyZADn38ebt2kn2wo9cAXU?via=matrix.org&via=element.io&via=beeper.com :heart:

Discussion being tracked in https://github.com/matrix-org/synapse/pull/13801#discussion_r970006848